python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -325,18 +330,22 @@ def __init__(
             raise NotImplementedError(
                 "Read from cloud storage"
             )  # pragma: no cover; no test yet
-        if any(
-            str(p).startswith("https:/" if POLARS_VERSION_LT_131 else "https://")
-            for p in self.paths
-        ):
+        if (
+            any(str(p).startswith("https:/") for p in self.paths)
+            and POLARS_VERSION_LT_131
+        ):  # pragma: no cover; no test yet
             raise NotImplementedError("Read from https")
         if any(
             str(p).startswith("file:/" if POLARS_VERSION_LT_131 else "file://")
             for p in self.paths
         ):
-            # TODO: removing the file:// may work
             raise NotImplementedError("Read from file URI")
         if self.typ == "csv":
+            if any(self._is_url(p) for p in self.paths):
+                # cannot because we do some file introspection
```

Comment:
After reading this, I'm not sure exactly what would need to be done in order to support reading remote CSV files. Could you either expand a bit on this, or open an issue with a description and link to it?